lib/mpl_toolkits/mplot3d/axis3d.py

```diff
@@ -178,23 +178,35 @@ def get_rotate_label(self, text):
             return len(text) > 4
 
     def _get_coord_info(self, renderer):
-        mins, maxs = np.array([
-            self.axes.get_xbound(),
-            self.axes.get_ybound(),
-            self.axes.get_zbound(),
-        ]).T
-        centers = (maxs + mins) / 2.
-        deltas = (maxs - mins) / 12.
-        mins = mins - deltas / 4.
-        maxs = maxs + deltas / 4.
-
-        vals = mins[0], maxs[0], mins[1], maxs[1], mins[2], maxs[2]
-        tc = self.axes.tunit_cube(vals, self.axes.M)
-        avgz = [tc[p1][2] + tc[p2][2] + tc[p3][2] + tc[p4][2]
-                for p1, p2, p3, p4 in self._PLANES]
-        highs = np.array([avgz[2*i] < avgz[2*i+1] for i in range(3)])
-
-        return mins, maxs, centers, deltas, tc, highs
+        mins, maxs = np.array(
+            [
+                self.axes.get_xbound(),
+                self.axes.get_ybound(),
+                self.axes.get_zbound(),
+            ]
+        ).T
+
+        # Get the mean value for each bound:
+        centers = 0.5 * (maxs + mins)
+
+        # Add a small offset between min/max point and the edge of the
+        # plot:
+        deltas = (maxs - mins) / 12.0
+        mins -= 0.25 * deltas
+        maxs += 0.25 * deltas
+
+        # Project the bounds along the current position of the cube:
+        bounds = mins[0], maxs[0], mins[1], maxs[1], mins[2], maxs[2]
+        bounds_proj = self.axes.tunit_cube(bounds, self.axes.M)
+
+        # Determine which one of the parallel planes are higher up:
+        highs = np.zeros(3, dtype=bool)
+        for i in range(3):
+            mean_z0 = np.mean(bounds_proj[self._PLANES[2 * i], 2])
+            mean_z1 = np.mean(bounds_proj[self._PLANES[2 * i + 1], 2])
+            highs[i] = mean_z0 < mean_z1
```

Comment:
Are we sure this gets tested?  I agree the previous code was very hard to follow, so I support this change if its identical.

Reply:
It's basically this without all the tricky for loops:
```python
        avgz = [tc[p1][2] + tc[p2][2] + tc[p3][2] + tc[p4][2]
                     for p1, p2, p3, p4 in self._PLANES]
        highs = np.array([avgz[2*i] < avgz[2*i+1] for i in range(3)])
```

The `highs` bool determines when to switch upside down or not so it pretty noticeable if it wouldn't have worked. 

You are correct though that it is not exactly the same values. `avgz` sums them and I used `mean` instead because that's what the name `avgz` seems to imply, it doesn't matter though if 

`(x0, x1, x2, x3) < (y0, y1, y2, y3)` 

is true then 

`(x0, x1, x2, x3)/4 < (y0, y1, y2, y3)/4` 

will also remain true.

I added some prints, but with np.sum instead for easy comparison:
```python
        avgz = [tc[p1][2] + tc[p2][2] + tc[p3][2] + tc[p4][2]
                     for p1, p2, p3, p4 in self._PLANES]
        highs = np.array([avgz[2 * i] < avgz[2 * i + 1] for i in range(3)])

        print(avgz)
        print("nextsdf")

        # Determine which one of the parallel planes are higher up:
        highs = np.zeros(3, dtype=bool)
        for i in range(3):
            mean_z0 = np.sum(bounds_proj[self._PLANES[2 * i], 2])
            mean_z1 = np.sum(bounds_proj[self._PLANES[2 * i + 1], 2])
            print(mean_z0, mean_z1)
            highs[i] = mean_z0 < mean_z1

Results:
[-3.922656332439214, -4.1027538546066875, -3.833147670696861, -4.19226251634904, -3.908760756651483, -4.1166494303944186]
nextsdf
-3.922656332439214 -4.1027538546066875
-3.833147670696861 -4.19226251634904
-3.908760756651483 -4.1166494303944186
```

I find comparing a mean value makes more sense than comparing the summed value. But np.sum is probably a little faster.

Reply:
No yours is definitely easier to read, and I can't imagine it is noticeably slower.